source/adapters/level_zero/usm.cpp

```diff
@@ -332,66 +333,17 @@ ur_result_t urUSMHostAlloc(
     size_t Size,
     /// [out] pointer to USM host memory object
     void **RetMem) {
-
-  uint32_t Align = USMDesc ? USMDesc->align : 0;
-  // L0 supports alignment up to 64KB and silently ignores higher values.
-  // We flag alignment > 64KB as an invalid value.
-  // L0 spec says that alignment values that are not powers of 2 are invalid.
-  // If alignment == 0, then we are allowing the L0 driver to choose the
-  // alignment so no need to check.
-  if (Align > 0) {
-    if (Align > 65536 || (Align & (Align - 1)) != 0)
-      return UR_RESULT_ERROR_INVALID_VALUE;
-  }
-
-  ur_platform_handle_t Plt = Context->getPlatform();
-  // If indirect access tracking is enabled then lock the mutex which is
-  // guarding contexts container in the platform. This prevents new kernels from
-  // being submitted in any context while we are in the process of allocating a
-  // memory, this is needed to properly capture allocations by kernels with
-  // indirect access. This lock also protects access to the context's data
-  // structures. If indirect access tracking is not enabled then lock context
-  // mutex to protect access to context's data structures.
-  std::shared_lock<ur_shared_mutex> ContextLock(Context->Mutex,
-                                                std::defer_lock);
-  std::unique_lock<ur_shared_mutex> IndirectAccessTrackingLock(
-      Plt->ContextsMutex, std::defer_lock);
-  if (IndirectAccessTrackingEnabled) {
-    IndirectAccessTrackingLock.lock();
-    // We are going to defer memory release if there are kernels with indirect
-    // access, that is why explicitly retain context to be sure that it is
-    // released after all memory allocations in this context are released.
-    UR_CALL(ur::level_zero::urContextRetain(Context));
-  } else {
-    ContextLock.lock();
-  }
-
-  // There is a single allocator for Host USM allocations, so we don't need to
-  // find the allocator depending on context as we do for Shared and Device
-  // allocations.
-  umf_memory_pool_handle_t hPoolInternal = nullptr;
+  ur_usm_pool_handle_t USMPool = nullptr;
   if (!UseUSMAllocator) {
```

Comment:
Isn't UseUSMAllocator a global variable? Can't we just read it in contextCreate and decide whether to create ProxyPool or DisjointPool there? We would only have one defaultPool in the context this way.

Reply:
Done. I've removed the `ProxyPool` from Context, now it's the `DefaultPool` that uses different pool implementation depending on the `UseUSMAllocator` varaible.